LanguageFeatures/Class-modifiers/versioning_A04_t01.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Going through a pre-feature library does not remove transitive
+/// restrictions for code in post-feature libraries.
+///
+/// @description Checks that it is a compile-time error if a declaration from a
+/// post-future library extends a platform library class marked `interface` and
+/// the superinterface chain goes through a pre-feature library declaration
+/// @author sgrekhov22@gmail.com
+
+import "versioning_pre_feature_lib.dart";
+
+class ExtendsPreFeatureImplementsInterface extends PreFeatureImplementsInterface {}
```

Comment:
I don't think these should be errors.

The class `PreFeatureImplementsInterface` is actually a perfectly good class also in the new language, because `implements InterfaceClass` does not imply any constraints (you don't have to have any particular class modifiers at all on a class that `implements InterfaceClass`).

So the migration of `PreFeatureImplementsInterface` will be a no-op with respect to class modifiers, and there is no need to flag these classes as errors.

Reply:
I guess the point is that restrictions associated with `interface` aren't transitive in the first place.